### ISSUE TITLE
2 warning seem to make training worng

### ISSUE THREAD

**Issue by TigerHH6866** [kohya-ss/sd-scripts#911]
because max_grad_norm is set, clip_grad_norm is enabled. consider set to 0 / max_grad_normが設定されているためclip_grad_normが有効になります。0に設定して無効にしたほうがいいかもしれません
running training / 学習開始
  num train images * repeats / 学習画像の数×繰り返し回数: 1050
  num reg images / 正則化画像の数: 0
  num batches per epoch / 1epochのバッチ数: 525
  num epochs / epoch数: 12
  batch size per device / バッチサイズ: 2
  gradient accumulation steps / 勾配を合計するステップ数 = 1
  total optimization steps / 学習ステップ数: 6300
steps:   0%|                                                                                                                                                                | 0/6300 [00:00<?, ?it/s]
epoch 1/12
/root/miniconda3/lib/python3.10/site-packages/torch/utils/checkpoint.py:31: UserWarning: None of the inputs have requires_grad=True. Gradients will be None
  warnings.warn("None of the inputs have requires_grad=True. Gradients will be None")

first one: max_grad_norm or clip_grad_norm  where to set?
seccond one: None of the inputs have requires_grad=True. Gradients will be None
this seem to make result worng. how to set right??

**Comment by TigerHH6866** [comment]
> > first one: max_grad_norm or clip_grad_norm where to set?
> 
> The default value of the max grad norm is `1`. You can set `0` with `--max_grad_norm` option.
> 
> > seccond one: None of the inputs have requires_grad=True. Gradients will be None
> > this seem to make result worng. how to set right??
> 
> This seems to be a false warning.
> 
> The first parameter of the model has a gradient by specifying `requires_grad_(True)` to enable the gradient checkpointing, but when we train LoRA, we do not train that parameter, so this warning seems to appear.

THX so much, it get goooooood training result now, Love you

**Comment by QiuLL** [comment]
I have the same issue,but get loss nan
use 8-bit AdamW optimizer | {}
running training / 学習開始
  num train images * repeats / 学習画像の数×繰り返し回数: 800
  num reg images / 正則化画像の数: 0
  num batches per epoch / 1epochのバッチ数: 800
  num epochs / epoch数: 7
  batch size per device / バッチサイズ: 1
  gradient accumulation steps / 勾配を合計するステップ数 = 1
  total optimization steps / 学習ステップ数: 5000
steps:   0%|                                                                                                                                                                              | 0/5000 [00:00<?, ?it/s]
epoch 1/7
/data/miniconda/envs/py10/lib/python3.10/site-packages/torch/utils/checkpoint.py:31: UserWarning: None of the inputs have requires_grad=True. Gradients will be None
  warnings.warn("None of the inputs have requires_grad=True. Gradients will be None")
/data/miniconda/envs/py10/lib/python3.10/site-packages/xformers/ops/fmha/flash.py:338: UserWarning: TypedStorage is deprecated. It will be removed in the future and UntypedStorage will be the only storage class. This should only matter to you if you are using storages directly.  To access UntypedStorage directly, use tensor.untyped_storage() instead of tensor.storage()
  and inp.query.storage().data_ptr() == inp.key.storage().data_ptr()
steps:   8%|████████████▏                                                                                                                                             | 397/5000 [02:22<27:29,  2.79it/s, loss=nan
